sdk/appconfiguration/app-configuration/samples/optimisticConcurrencyViaEtag.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// This sample shows how to implement optimistic concurrency using 
+// App Configuration and etags.
+
+// NOTE: replace with import { AppConfigurationClient } from "@azure/app-configuration"
+// in a standalone project
+import { AppConfigurationClient } from "../src";
+
+export async function run() {
+  console.log("Running optimistic concurrency sample");
+
+  // You will need to set this environment variable
+  const connectionString = process.env["AZ_CONFIG_CONNECTION"]!;
+  const client = new AppConfigurationClient(connectionString);
+
+  const key = "optimisticConcurrencySampleKey";
+  await cleanupSampleValues([key], client);
+
+  console.log("Creating a new key");
+  const setting = await client.addConfigurationSetting({ key: key, value: "Initial value" });
+  console.log(`Etag for ${key} is ${setting.etag}`);
+
+  // Each setting (on update or add) gets a unique identifier, via the .etag property.
+  //
+  // This etag can be passed to the server when retrieving or updating settings to prevent
+  // race conditions where two independent updaters keep overwriting each other.
+  // 
+  // https://en.wikipedia.org/wiki/Optimistic_concurrency_control#Web_usage
+
+  // let's simulate this with two updaters - Alpha and Beta
+  console.log("Starting Alpha and Beta updates");
+
+  // Alpha is preparing to do an update - to do this properly they retrieve the setting from the server
+  let alphaSetting = await client.getConfigurationSetting(key);
+  
+  // Beta would also like to do an update and will do the same thing as Alpha
+  let betaSetting = await client.getConfigurationSetting(key);
+  
+  console.log(`Alpha retrieves the setting and has etag ${alphaSetting.etag}`);
+  console.log(`Beta retrieves the setting and has etag ${betaSetting.etag}`);
+
+  // Since both of Alpha and Beta will attempt to update at the same time
+  // they'll have to coordinate this. This is when they use the etag.
+  
+  // For our example we'll have Beta update first
+  const betaUpdatedSetting = await client.setConfigurationSetting({
+    key: key,
```

Comment:
`key`

Reply:
(explained in other comment - purposeful verbosity)